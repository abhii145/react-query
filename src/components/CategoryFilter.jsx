
const CategoryFilter = ({ categories, category, onCategoryChange }) => (
  <div className="flex items-center gap-2">
    <label htmlFor="category" className="text-sm font-medium text-gray-600 whitespace-nowrap">
      Category:
    </label>
    <select
      id="category"
      value={category}
      onChange={onCategoryChange}
      className="p-2 border border-gray-300 rounded-md text-sm focus:ring-2 focus:ring-blue-500 focus:outline-none"
    >
      <option value="All">All</option>
      {categories?.map((category) => (
        <option key={category.slug} value={category.slug}>
          {category.name}
        </option>
      ))}
    </select>
  </div>
)

export default CategoryFilter