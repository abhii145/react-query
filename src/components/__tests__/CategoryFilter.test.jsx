import { render, screen, fireEvent } from "@testing-library/react"
import CategoryFilter from "../CategoryFilter"
import { test, expect, describe, vi } from "vitest"

const categories = [
  { slug: "electronics", name: "Electronics" },
  { slug: "fashion", name: "Fashion" },
]

describe("CategoryFilter component", () => {
  test("renders CategoryFilter component", () => {
    const handleCategoryChange = vi.fn()

    render(
      <CategoryFilter
        categories={categories}
        category="All"
        onCategoryChange={handleCategoryChange}
      />
    )

    expect(screen.getByLabelText(/Category:/i)).toBeInTheDocument()
    expect(screen.getByRole("combobox")).toHaveValue("All")
    expect(screen.getByRole("option", { name: "All" })).toBeInTheDocument()
    expect(
      screen.getByRole("option", { name: "Electronics" })
    ).toBeInTheDocument()
    expect(screen.getByRole("option", { name: "Fashion" })).toBeInTheDocument()
  })

  test("calls onCategoryChange when a new category is selected", () => {
    const handleCategoryChange = vi.fn()

    render(
      <CategoryFilter
        categories={categories}
        category="All"
        onCategoryChange={handleCategoryChange}
      />
    )

    fireEvent.change(screen.getByRole("combobox"), {
      target: { value: "electronics" },
    })
    expect(handleCategoryChange).toHaveBeenCalled()
  })
})
