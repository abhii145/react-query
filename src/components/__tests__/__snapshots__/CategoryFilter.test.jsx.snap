// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CategoryFilter component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex items-center gap-2"
  >
    <label
      class="text-sm font-medium text-gray-600 whitespace-nowrap"
      for="category"
    >
      Category:
    </label>
    <select
      class="p-2 border border-gray-300 rounded-md text-sm focus:ring-2 focus:ring-blue-500 focus:outline-none"
      id="category"
    >
      <option
        value="All"
      >
        All
      </option>
      <option
        value="electronics"
      >
        Electronics
      </option>
      <option
        value="fashion"
      >
        Fashion
      </option>
    </select>
  </div>
</DocumentFragment>
`;
